pkg/gvkcompare/gvkcompare.go

```diff
@@ -0,0 +1,276 @@
+package gvkcompare
+
+import (
+	"sort"
+	"strings"
+
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/discovery"
+)
+
+// GVKCompare reconciles a GVKCompare object
+type GVKCompare struct {
```

Comment:
Within the `gvk` package, suggest the type named `Comparator` would be intuitive.  Would be used as: `gvk.Comparator{}`